go/vt/vttablet/tabletmanager/vdiff/stats_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vdiff
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/stats"
+
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+)
+
+func TestVDiffStats(t *testing.T) {
+	testStats := &vdiffStats{
+		ErrorCount:          stats.NewCounter("", ""),
+		RestartedTableDiffs: stats.NewCountersWithSingleLabel("", "", "Table", ""),
+		RowsDiffedCount:     stats.NewCounter("", ""),
+	}
+	id := int64(1)
+	testStats.controllers = map[int64]*controller{
+		id: {
+			id:                    id,
+			workflow:              "testwf",
+			workflowType:          binlogdatapb.VReplicationWorkflowType_MoveTables,
+			uuid:                  uuid.New().String(),
+			Errors:                stats.NewCountersWithMultiLabels("", "", []string{"Error"}),
+			TableDiffRowCounts:    stats.NewCountersWithMultiLabels("", "", []string{"Rows"}),
+			TableDiffPhaseTimings: stats.NewTimings("", "", "", "TablePhase"),
+		},
+	}
+
+	require.Equal(t, int64(1), testStats.numControllers())
+
+	sleepTime := 1 * time.Millisecond
+	record := func(phase string) {
+		defer testStats.controllers[id].TableDiffPhaseTimings.Record(phase, time.Now())
+		time.Sleep(sleepTime)
+	}
+	want := int64(1.2 * float64(sleepTime)) // Allow 20% overhead for recording timing
```

Comment:
We have seen flakiness with such precise timings in the past, especially since you are dealing in milliseconds here. We should probably just compare that it is not zero.

Reply:
I can update it if it becomes flaky. I don't recall ever seeing `TestVReplicationStats` fail, and that does the same thing. If they are flaky then we should update both.